client/packages/system/src/Stock/Components/Ledger/useLedgerColumns.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  ColumnFormat,
+  useColumns,
+  useFormatDateTime,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { LedgerLine } from '../../api/hooks/useStockLedger';
+
+type MovementType =
+  | 'INBOUND_SHIPMENT'
+  | 'OUTBOUND_SHIPMENT'
+  | 'INBOUND_RETURN'
+  | 'OUTBOUND_RETURN'
+  | 'PRESCRIPTION'
+  | 'INVENTORY_ADDITION'
+  | 'INVENTORY_REDUCTION';
+
+export const useLedgerColumns = () => {
+  const t = useTranslation('app');
+  const { localisedTime } = useFormatDateTime();
+
+  const columns = useColumns<LedgerLine>([
+    {
+      key: 'datetime',
+      label: 'label.date',
+      format: ColumnFormat.Date,
+    },
+    {
+      key: 'time',
+      label: 'label.time',
+      accessor: ({ rowData }) => localisedTime(rowData.datetime),
+    },
+    {
+      key: 'name',
+      label: 'label.name',
+    },
+    {
+      key: 'quantity',
+      label: 'label.quantity',
+    },
+    {
+      key: 'type',
+      label: 'label.type',
+      accessor: ({ rowData }) =>
+        t(getLocalisationKey(rowData.type as MovementType)),
+    },
+    {
+      key: 'reason',
+      label: 'label.reason',
+    },
+  ]);
+
+  return { columns };
+};
+
+const getLocalisationKey = (type: MovementType) => {
+  switch (type) {
```

Comment:
Would be great to add exhaustive match here:

https://github.com/msupply-foundation/open-msupply/blob/c9fa2d2a3b79d90cd8cd57df91b62bbf43e7fb50/client/packages/host/src/components/Sync/Sync.tsx#L89-L90

Btw even without the API implemented you can use [InvoiceNodeType](https://github.com/msupply-foundation/open-msupply/blob/c9fa2d2a3b79d90cd8cd57df91b62bbf43e7fb50/client/packages/common/src/types/schema.ts#L2726-L2735), also sounds like we need repack type too

Reply:
Done. I haven't included "REPACK" in the back-end View query -- should I?

Reply:
Okay, have added to back-end PR now